File: AndroidApp/app/src/main/java/com/tv2oppgave/tvmovieapp/data/models/Feed.kt
package com.tv2oppgave.tvmovieapp.data.models

data class FeedContentItem(
    val id: String,
    val title: String,
    val imageUrl: String?,
    val urlPath: String
)

data class FeedResponse(
    val id: String,
    val title: String,
    val section_title: String,
    val movies: List<FeedContentItem>
)

File: WebApp/src/models/feed.ts
export interface RootFeedResponse {
  layout: string;
  feeds: Feed[];
}

export interface Feed {
  id: string;
  title: string;
  section_title?: string;
  type: string;
  size: number;
  start: number;
  total: number;

  styles?: {
    layout?: {
      name: string;
      text?: string;
      aspect?: string;
    };
    theme?: {
      name: string;
    };
  };

  movies: FeedMovie[];
}

export interface FeedContentItem {
  content_id: string;
  title: string;
  description?: string;
  image: {
    src: string;
  };
  url: string;
}

export interface FeedMovie {
  id: string;
  title: string;
  imageUrl: string;
  urlPath: string;
}

export interface Label {
  text: string;
  type: string;
}

